Update endpoints archetypes with new tooling

frankyn opened the pull request:
* Added Endpoints Frameworks v2.0 support
* Added gradle build support to generated archetype (this is not gradle archetype generation)
* Added support for new AppEngine plugins 
* Added comments on how to use Endpoints Frameworks v1.0 support. It's commented unless developer wants to use this framework. 
* Updated README.md to reflect changes with new tooling
* Fixed bug where archetype variables weren't replaced when archetype is generated in certain files.
* Updated archetype version to 2.0.0-0 following maven convention. 
* Added Jenkins End-2-End testing for hello-endpoints-archetype

Commit:
Updating endpoints tooling

* Added gradle build support
* Added endpoints v1.0 and v2.0 support to archetype

Commit:
Moving and changing copyright header

Commit:
Moving copyright below xml header...

Commit:
Adding Test for hello-endpoints-archetype

Commit:
Adding Gradle appengineDeploy to E2E tests

Commit:
Updating Jenkins.sh to fail on non-zero return

Commit:
Removing unnecessary paramters application/version

Commit:
Removing unnecessary application/version parameters

Commit:
Added -x to Jenkins.sh

Commit:
Add fail on non-zero and print commands to Jenkins.sh

Commit:
Set deployment version for gradle test

Commit:
Removing unnecessary setup step from README

lesv commented on endpoints-skeleton-archetype/pom.xml:
```diff
@@ -17,18 +17,18 @@ limitations under the License.
   <groupId>com.google.appengine.archetypes</groupId>
   <artifactId>endpoints-skeleton-archetype</artifactId>
-  <version>1.1.5-${appengine.sdk.version}</version>
+  <version>2.0.0-0</version>
```
Let's loose the `-0`

lesv commented on endpoints-skeleton-archetype/src/main/resources/META-INF/maven/archetype-metadata.xml:
```diff
@@ -30,22 +30,26 @@
       <includes>
         <include>LICENSE</include>
         <include>README.md</include>
         <include>nbactions.xml</include>
```
It might make sense to loose nbactions.xml

lesv commented on endpoints-skeleton-archetype/src/main/resources/archetype-resources/pom.xml:
```diff
@@ -3,6 +3,21 @@
 #set( $symbol_escape = '\' )
 <?xml version="1.0" encoding="UTF-8"?>
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+<!--
```
We don't put a copyright here --- it's user generated code.

lesv commented on endpoints-skeleton-archetype/src/main/resources/archetype-resources/pom.xml:
```diff
@@ -96,18 +152,24 @@
+                </configuration>
+            </plugin>
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
```
Not required

lesv commented on endpoints-skeleton-archetype/src/main/resources/archetype-resources/pom.xml:
```diff
@@ -96,18 +152,24 @@
             </plugin>
-
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
```
Not required.

lesv commented on endpoints-skeleton-archetype/src/main/resources/archetype-resources/pom.xml:
```diff
@@ -96,18 +152,24 @@
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-war-plugin</artifactId>
-                <version>2.6</version>
+                <version>${maven.war.version}</version>
```
Probably not required

lesv commented on endpoints-skeleton-archetype/src/main/resources/archetype-resources/src/main/webapp/WEB-INF/web.xml:
```diff
@@ -2,17 +2,52 @@
 <?xml version="1.0" encoding="utf-8" standalone="no"?><web-app xmlns="http://java.sun.com/xml/ns/javaee" xmlns:web="http://java.sun.com/xml/ns/javaee/web-app_2_5.xsd" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" version="2.5" xsi:schemaLocation="http://java.sun.com/xml/ns/javaee http://java.sun.com/xml/ns/javaee/web-app_2_5.xsd">
+<!--
+Copyright 2017 Google Inc. All Rights Reserved.
+
```
No copyright in generated code.

lesv commented on hello-endpoints-archetype/src/main/resources/archetype-resources/build.gradle:
```diff
@@ -0,0 +1,70 @@
+#set( $symbol_pound = '#' )
+#set( $symbol_dollar = '$' )
+#set( $symbol_escape = '\' )
+// Copyright 2017 Google Inc. All Rights Reserved.
```
No copyright required here.

lesv commented on hello-endpoints-archetype/src/main/resources/archetype-resources/eclipse-launch-profiles/DevAppServer.launch:
```diff
@@ -1,10 +1,7 @@
 <?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
We may wish to get rid of these.

lesv commented on hello-endpoints-archetype/src/main/resources/archetype-resources/pom.xml:
```diff
@@ -1,24 +1,52 @@
+#set( $symbol_escape = '\' )
 <?xml version="1.0" encoding="UTF-8"?>
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+<!--
```
remove copyright here.

lesv commented on hello-endpoints-archetype/src/main/resources/archetype-resources/pom.xml:
```diff
@@ -1,24 +1,52 @@
 
-    <groupId>${groupId}</groupId>
-    <artifactId>${artifactId}</artifactId>
+    <groupId>com.example.helloendpoints</groupId>
```
This should be generated by user input.

lesv commented on hello-endpoints-archetype/src/main/resources/archetype-resources/pom.xml:
```diff
@@ -1,24 +1,52 @@
-    <groupId>${groupId}</groupId>
-    <artifactId>${artifactId}</artifactId>
+    <groupId>com.example.helloendpoints</groupId>
+    <artifactId>helloendpoints</artifactId>
```
This should be a `${...}`  (ie the user sets it.)

lesv commented on hello-endpoints-archetype/src/main/resources/archetype-resources/src/main/webapp/WEB-INF/web.xml:
```diff
@@ -2,17 +2,52 @@
 #set( $symbol_escape = '\' )
 <?xml version="1.0" encoding="utf-8" standalone="no"?><web-app xmlns="http://java.sun.com/xml/ns/javaee" xmlns:web="http://java.sun.com/xml/ns/javaee/web-app_2_5.xsd" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" version="2.5" xsi:schemaLocation="http://java.sun.com/xml/ns/javaee http://java.sun.com/xml/ns/javaee/web-app_2_5.xsd">
+<!--
+Copyright 2017 Google Inc. All Rights Reserved.
```
Remove Copyright notice.

lesv reviewed (changes requested):
Remove Copyright notices in generated code.
Not sure where app.id is used.  Shouldn't be required anymore.

Commit:
Removed eclipse-launch-profiles

Commit:
Updated archetype version - removed "-0"

Commit:
Removed -0 from helloendpoints pom.

Commit:
Remove nbactions and eclipse profiles

Commit:
Updating copyright in hellendpoints and endpointsskeleton

* Removed copyright from auto-generated code

Commit:
Modified velocity commented copyrights - remove whitespace.

Commit:
Removed unnecessary plugins from endpoints archetypes pom.xml

Commit:
Updating Jenkins.sh to be more specific to running instance

Commit:
Updating READMEs for endpoints archetypes

Commit:
Modifying PROJECT_ID to VERSION_ID in Jenkins.sh tests

Commit:
Consolidated Jenkins.sh and archetype/Jenkins.sh

frankyn commented:
@lesv PTAL

1. Acknowledging comments.
2. Adding a test to verify specific deployed version versus a prior deployment. 
3. Updated static pom values
```
<groupId>com.example.helloendpoints</groupId>
<artifactId>helloendpoints</artifactId>
```
4. Updated Copyrights to be Velocity comments
5. Updated README.md to reflect our conversation. 
6. Added e2e test to Jenkins

lesv commented on endpoints-skeleton-archetype/src/main/resources/archetype-resources/build.gradle:
```diff
@@ -0,0 +1,72 @@
+  }
+  dependencies {
+    classpath 'com.google.cloud.tools:appengine-gradle-plugin:+'    // latest App Engine Gradle tasks
+    classpath 'com.google.cloud.tools:endpoints-framework-gradle-plugin:${endpoints-p}'
```
Wow - does that really work?

lesv commented:
So the same comments I gave about the sample should apply here, except, of course, the parent pom doesn't appear in the generated code.  (Nor do we have copyrights).

Commit:
Updating endpoints archetypes READMEs with note for payment.

Commit:
Cleand up pom.xml and removed unnecessary dependency

Commit:
Updated Jenkins.sh to be version specific.

frankyn commented:
@Les, I modified archetypes to use velocity comments as suggested from a prior meeting. Do you want to remove copyright headers from the archetypes?

Commit:
Updating skeleton to act more like an endpoints skeleton

Commit:
Updating skeleton endpoints archetype README to reflect changes

Commit:
Updating README for helloendpoints archetype

Commit:
Removing unused application-id from endpoints archetypes

Commit:
Adding header for endpoints archetypes README

Commit:
Wrapped environment variables with double-quotes

frankyn commented:
@lesv PTAL
